lib/Service/Social/MastodonProvider.php

```diff
@@ -104,14 +104,18 @@ protected function getProfileIds($contact):array {
 					if (isset($masto_user_server)) {
 						try {
 							[$masto_user, $masto_server] = $masto_user_server;
-							# search for user on Mastodon
-							$search = $masto_server . '/api/v2/search?q=' . $masto_user . '@' . parse_url($masto_server)["host"];
-							$result = $this->httpClient->get($search);
+							# search for user webfinger
+							$webfinger = $masto_server . '/.well-known/webfinger?resource=acct:' . $masto_user . '@' . parse_url($masto_server)["host"];
+							$result = $this->httpClient->get($webfinger);
 							$jsonResult = json_decode($result->getBody());
```

Comment:
**suggestion:** 
```suggestion
							$jsonResult = json_decode($result->getBody(), null, 512,  JSON_THROW_ON_ERROR);
```
So that JSON parsing error is thrown as well, instead of returning `null`